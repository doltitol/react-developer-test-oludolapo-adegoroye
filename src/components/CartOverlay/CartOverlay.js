import React, { PureComponent } from 'react';
import { numberCommaFormatter } from '../../util';
import MainButton from '../Buttons/MainButton';
import CartOverlayItem from '../CartOverlayItem/CartOverlayItem';
import './cartoverlay.style.scss';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

export class CartOverlay extends PureComponent {
  getTotal = () => {
    const cartTotal = this.props.cartItems.reduce(
      (acc, cur) =>
        acc +
        cur.prices.filter(
          (price) => price.currency.symbol === this.props.currency
        )[0].amount *
          cur.qty,
      0
    );
    return { total: cartTotal };
  };
  render() {
    return (
      <div data-testid='cart-overlay'>
        <div
          className='cart-overlay'
          onClick={() => this.props.closeCart()}
        ></div>
        <div className='cart-overlay-container slide-right'>
          <div className='cart-overlay-container-header'>
            <p className='cart-overlay-container-header-text'>
              <strong>My Bag</strong>, {this.props.cartQuantity} items
            </p>
          </div>
          <div className='cart-overlay-container-cart'>
            {this.props.cartItems.map((item, index) => (
              <CartOverlayItem
                cartItem={item}
                key={`${item.id}-${index}`}
                closeCart={() => this.props.closeCart()}
              />
            ))}
          </div>
          <div className='cart-overlay-container-total'>
            <p className='cart-overlay-container-total-text'>Total</p>
            <p className='cart-overlay-container-total-pricing'>
              {this.props.currency}
              {numberCommaFormatter(this.getTotal().total.toFixed(2))}
            </p>
          </div>
          <div className='cart-overlay-container-buttons'>
            <Link to={'/cart'} onClick={() => this.props.closeCart()}>
              <MainButton text='view bag' width='135px' height='43px' outline />
            </Link>
            <MainButton text='check out' width='135px' height='43px' />
          </div>
        </div>
      </div>
    );
  }
}
const mapStateToProps = (state) => {
  return {
    cartItems: state.cart.cartItems,
    currency: state.cart.currency,
    cartQuantity: state.cart.cartQuantity,
    total: state.cart.total,
  };
};

export default connect(mapStateToProps)(CartOverlay);
