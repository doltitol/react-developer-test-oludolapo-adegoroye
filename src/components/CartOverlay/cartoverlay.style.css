@import url("https://fonts.googleapis.com/css2?family=Raleway:wght@400;600;700&family=Roboto+Condensed:wght@700&family=Roboto:wght@400;500&family=Source+Sans+Pro&display=swap");
* {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  padding: 0;
  margin: 0;
  scroll-behavior: smooth;
}

body {
  background: #ffffff;
}

a {
  text-decoration: none;
}

.slide-right {
  -webkit-animation: slide-right 0.5s cubic-bezier(0.25, 0.46, 0.45, 0.94) both;
  animation: slide-right 0.5s cubic-bezier(0.25, 0.46, 0.45, 0.94) both;
}

@-webkit-keyframes slide-right {
  0% {
    -webkit-transform: translateX(50px);
    transform: translateX(50px);
  }
  100% {
    -webkit-transform: translateX(0px);
    transform: translateX(0px);
  }
}

@keyframes slide-right {
  0% {
    -webkit-transform: translateX(50px);
    transform: translateX(50px);
  }
  100% {
    -webkit-transform: translateX(0px);
    transform: translateX(0px);
  }
}

.slide-down {
  -webkit-animation: slide-down 0.5s cubic-bezier(0.25, 0.46, 0.45, 0.94) both;
  animation: slide-down 0.5s cubic-bezier(0.25, 0.46, 0.45, 0.94) both;
}

@-webkit-keyframes slide-down {
  0% {
    -webkit-transform: translateY(-25%);
    transform: translateY(-25%);
  }
  100% {
    -webkit-transform: translateY(0px);
    transform: translateY(0px);
  }
}

@keyframes slide-down {
  0% {
    -webkit-transform: translateY(-25%);
    transform: translateY(-25%);
  }
  100% {
    -webkit-transform: translateY(0px);
    transform: translateY(0px);
  }
}

.cart-overlay {
  position: fixed;
  /* Sit on top of the page content */
  width: 100%;
  /* Full width (cover the whole page) */
  height: 100%;
  /* Full height (cover the whole page) */
  top: 70px;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(57, 55, 72, 0.22);
  /* Black background with opacity */
  z-index: 2;
  /* Specify a stack order in case you're using a different order for other elements */
  cursor: pointer;
  /* Add a pointer on hover */
  z-index: 20;
}

.cart-overlay-container {
  width: 325px;
  height: 677px;
  background-color: #ffffff;
  padding: 30px 20px;
  position: absolute;
  top: 80px;
  right: 70px;
  z-index: 21;
}

.cart-overlay-container-header-text {
  font-family: "Raleway", sans-serif;
  font-weight: 400;
  color: #1D1F22;
  font-size: 16px;
  line-height: 25.6px;
}

.cart-overlay-container-cart {
  height: 495px;
  padding: 0 10px;
  overflow: auto;
}

.cart-overlay-container-total {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}

.cart-overlay-container-total-text {
  font-family: "Roboto", sans-serif;
  font-weight: 500;
  color: #1D1F22;
  font-size: 16px;
  line-height: 18px;
}

.cart-overlay-container-total-pricing {
  font-family: "Raleway", sans-serif;
  font-weight: 700;
  color: #1D1F22;
  font-size: 16px;
  line-height: 25.6px;
}

.cart-overlay-container-buttons {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-direction: row;
          flex-direction: row;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
  margin-top: 30px;
}
/*# sourceMappingURL=cartoverlay.style.css.map */