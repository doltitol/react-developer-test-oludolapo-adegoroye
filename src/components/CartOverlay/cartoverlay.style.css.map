{
    "version": 3,
    "mappings": "ACAA,OAAO,CAAC,wKAAI;AASZ,AAAA,CAAC,CAAC;EACA,UAAU,EAAE,UAAU;EACtB,OAAO,EAAE,CAAC;EACV,MAAM,EAAE,CAAC;EACT,eAAe,EAAE,MAAM;CACxB;;AACD,AAAA,IAAI,CAAA;EACJ,UAAU,ECZF,OAAO;CDad;;AAED,AAAA,YAAY,CAAC;EACX,iBAAiB,EAAE,WAAW,CAAC,IAAI,CAAC,oCAAoC,CAAC,IAAI;EAC7E,SAAS,EAAE,WAAW,CAAC,IAAI,CAAC,oCAAoC,CAAC,IAAI;CACtE;;AAED,kBAAkB,CAAlB,WAAkB;EAChB,EAAE;IACA,iBAAiB,EAAE,gBAAgB;IACnC,SAAS,EAAE,gBAAgB;;EAE7B,IAAI;IACF,iBAAiB,EAAE,eAAe;IAClC,SAAS,EAAE,eAAe;;;;AAI9B,UAAU,CAAV,WAAU;EACR,EAAE;IACA,iBAAiB,EAAE,gBAAgB;IACnC,SAAS,EAAE,gBAAgB;;EAE7B,IAAI;IACF,iBAAiB,EAAE,eAAe;IAClC,SAAS,EAAE,eAAe;;;;ADzC9B,AAAA,aAAa,CAAC;EACZ,gBAAgB,EEMR,sBAAsB;EFL9B,OAAO,EAAE,MAAM;EACf,KAAK,EAAE,IAAI;EACX,MAAM,EAAE,kBAAkB;EAC1B,IAAI,EAAE,CAAC;EACP,KAAK,EAAE,CAAC;EACR,QAAQ,EAAE,QAAQ;EAClB,OAAO,EAAE,IAAI;EACb,GAAG,EAAE,IAAI;EACT,OAAO,EAAE,GAAG;EACZ,eAAe,EAAE,QAAQ;CA0C1B;;AAxCC,MAAM,EAAE,SAAS,EAAE,KAAK;EAb1B,AAAA,aAAa,CAAC;IAcV,OAAO,EAAE,MAAM;GAuClB;;;AArCC,MAAM,EAAE,SAAS,EAAE,KAAK;EAhB1B,AAAA,aAAa,CAAC;IAiBV,OAAO,EAAE,MAAM;GAoClB;;;AAlCE,AAAD,uBAAW,CAAC;EACV,KAAK,EAAE,KAAK;EACZ,MAAM,EAAE,KAAK;EACb,gBAAgB,EEnBZ,OAAO;EFoBX,OAAO,EAAE,SAAS;CA6BnB;;AA3BI,AAAD,mCAAM,CAAC;EGlBX,WAAW,EANC,SAAS,EAAE,UAAU;EAOjC,WAAW,EHkB0B,GAAG;EGjBxC,KAAK,EDRC,OAAO;ECSb,SAAS,EHgBc,IAAI;EGf3B,WAAW,EHekB,MAAM;CAC9B;;AAEF,AAAD,4BAAM,CAAC;EACL,MAAM,EAAE,KAAK;EACb,OAAO,EAAE,MAAM;EACf,QAAQ,EAAE,IAAI;CACf;;AACA,AAAD,6BAAO,CAAC;EACN,OAAO,EAAE,IAAI;EACb,eAAe,EAAE,aAAa;EAC9B,WAAW,EAAE,MAAM;CAOpB;;AANE,AAAD,kCAAM,CAAC;EGxBX,WAAW,EAZA,QAAQ,EAAE,UAAU;EAa/B,WAAW,EHwBuB,GAAG;EGvBrC,KAAK,EDfC,OAAO;ECgBb,SAAS,EHsBa,IAAI;EGrB1B,WAAW,EHqBiB,IAAI;CAC3B;;AACA,AAAD,qCAAS,CAAC;EGlCd,WAAW,EANC,SAAS,EAAE,UAAU;EAOjC,WAAW,EHkC0B,GAAG;EGjCxC,KAAK,EDRC,OAAO;ECSb,SAAS,EHgCc,IAAI;EG/B3B,WAAW,EH+BkB,MAAM;CAC9B;;AAEF,AAAD,+BAAS,CAAA;EACP,OAAO,EAAE,IAAI;EACb,cAAc,EAAE,GAAG;EACnB,WAAW,EAAE,MAAM;EACnB,eAAe,EAAE,aAAa;EAC9B,UAAU,EAAE,IAAI;CACjB",
    "sources": [
        "cartoverlay.style.scss",
        "../../styles/index.scss",
        "../../styles/_colors.scss",
        "../../styles/_fonts.scss",
        "../../styles/_colors.scss"
    ],
    "names": [],
    "file": "cartoverlay.style.css"
}