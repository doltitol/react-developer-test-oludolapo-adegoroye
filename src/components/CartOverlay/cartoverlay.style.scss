@import '../../styles/index.scss';
.cart-overlay {
 position: fixed; /* Sit on top of the page content */
  width: 100%; /* Full width (cover the whole page) */
  height: 100%; /* Full height (cover the whole page) */
  top: 70px;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: $overlay; /* Black background with opacity */
  z-index: 2; /* Specify a stack order in case you're using a different order for other elements */
  cursor: pointer; /* Add a pointer on hover */
  z-index: 20;

  @media (max-width: 960px) {
  
  }
  @media (max-width: 550px) {

  }
  &-container {
    width: 350px;
    height: 677px;
    background-color: $white;
    padding: 30px 20px;
    position: absolute;
    top: 70px;
    right: 70px;
    z-index: 21;
    &-header {
      &-text {
        @include raleway(16px, 25.6px, 400);
      }
    }
    &-cart {
      height: 495px;
      padding: 20px 0px;
      padding-right: 10px;
      overflow: auto;
         &::-webkit-scrollbar {
              width: 4px;
            }
            &::-webkit-scrollbar-track {
              box-shadow: inset 0 0 4px rgba(0, 0, 0, 0.05);
            }
            &::-webkit-scrollbar-thumb {
              background-color: $primary;
              outline: 1px solid $primary;
              border-radius: 8px;
            }
    }
    &-total {
      display: flex;
      justify-content: space-between;
      align-items: center;
      margin-top: 10px;
      &-text {
        @include roboto(16px, 18px, 500);
      }
      &-pricing {
        @include raleway(16px, 25.6px, 700);
      }
    }
    &-buttons{
      display: flex;
      flex-direction: row;
      align-items: center;
      justify-content: space-between;
      margin-top: 30px;
    }
  }
}
