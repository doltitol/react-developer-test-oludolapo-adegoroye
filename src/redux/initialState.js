export const initialState = {
  cartItems: JSON.parse(localStorage.getItem('cartItems') || '[]'),
  cartQuantity: JSON.parse(localStorage.getItem('cartItems') || '[]').reduce(
    (acc, item) => acc + item.qty,
    0
  ),
  currency: '$',
};
