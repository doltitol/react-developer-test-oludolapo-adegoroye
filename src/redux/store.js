import { createStore, combineReducers } from 'redux';
import cartReducer from './reducers/cartReducers';

const rootReducer = combineReducers({
  cart: cartReducer,
});
export const store = createStore(rootReducer);
