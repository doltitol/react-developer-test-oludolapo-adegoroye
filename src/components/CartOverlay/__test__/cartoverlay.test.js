import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../../../redux/store';
import CartOverlay from '../CartOverlay';

describe('<CartOverlay /> spec', () => {
  it('renders the empty component', () => {
    const container = render(
      <Provider store={store}>
        <BrowserRouter>
          <CartOverlay />
        </BrowserRouter>
      </Provider>
    );
    const getByTestId = container.getByTestId;
    const cartOverlay = getByTestId('cart-overlay');
    expect(cartOverlay.children).toHaveLength(2);
    expect(container.children).toMatchSnapshot();
  });
});
