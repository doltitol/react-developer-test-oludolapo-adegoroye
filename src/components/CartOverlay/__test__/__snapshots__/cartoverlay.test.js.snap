// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartOverlay /> spec renders the empty component 1`] = `undefined`;
